lib/prompb/prompb.go

```diff
@@ -34,6 +34,24 @@ func (wr *WriteRequest) Reset() {
 		samplesPool[i] = Sample{}
 	}
 	wr.samplesPool = samplesPool[:0]
+	exemplarsPool := wr.exemplarsPool
+	for i := range exemplarsPool {
+		exemplarsPool[i] = Exemplar{}
+	}
+	wr.exemplarsPool = exemplarsPool[:0]
+}
+
+// Exemplar is an exemplar
+type Exemplar struct {
+	// Labels a list of labels that uniquely identifies exemplar
+	// Optional, can be empty.
+	Labels []Label
+	// Value: the value of the exemplar
+	Value float64
+	// timestamp is in ms format, see model/timestamp/timestamp.go for
+	// conversion from time.Time to Prometheus timestamp.
+	Timestamp  int64
+	labelsPool []Label
```

Comment:
It'd be great to use a global labelsPool, instead of local per examplar.

 Since, actually  Examplar.Labels should be a reference to the writeReques.labelsPool via sub-slice. It allows to use less memory and reduce the number of allocations.

Reply:
Spot on.  I discovered this as well last night when I was cherry picking to the cluster branch.   labelsPool is being sent through to insert.  Ill modify and update.

Reply:
Refactored to use exemplarLabelsPool on WriteRequest because field order is not mandatory with protobuf